public/app/features/gops/configuration-tracker/hooks/irmHooks.ts

```diff
@@ -0,0 +1,380 @@
+}
+export interface SectionDtoStep {
+  title: string;
+  description: string;
+  button: StepButtonDto;
+}
+export interface SectionDto {
+  title: string;
+  description: string;
+  steps: SectionDtoStep[];
+}
+export interface SectionsDto {
+  sections: SectionDto[];
+}
+
+function isCreateAlertRuleDone() {
+  const { data: namespaces = [] } = alertRuleApi.endpoints.prometheusRuleNamespaces.useQuery(
+    {
+      ruleSourceName: GRAFANA_RULES_SOURCE_NAME,
+    },
+    {
+      refetchOnFocus: true,
+      refetchOnReconnect: true,
+      refetchOnMountOrArgChange: true,
+    }
+  );
+  return namespaces.length > 0;
+}
+
+function isContactPointReady(defaultContactPoint: string, contactPoints: Receiver[]) {
+  // We consider the contact point ready if the default contact has the address filled
+
+  const defaultEmailUpdated = contactPoints.some(
+    (contactPoint: Receiver) =>
+      contactPoint.name === defaultContactPoint &&
+      contactPoint.grafana_managed_receiver_configs?.some(
+        (receiver) => receiver.name === defaultContactPoint && receiver.settings?.address !== '<example@email.com>'
+      )
+  );
+  return defaultEmailUpdated;
+}
+
+function isOnCallContactPointReady(contactPoints: Receiver[]) {
+  return contactPoints.some((contactPoint: Receiver) =>
+    contactPoint.grafana_managed_receiver_configs?.some((receiver) => receiver.type === 'oncall')
+  );
+}
+
+interface IncidentsPluginConfig {
+  isInstalled: boolean;
+  isChatOpsInstalled: boolean;
+  isIncidentCreated: boolean;
+}
+
+function useGetContactPoints() {
+  const alertmanagerConfiguration = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useQuery(
+    GRAFANA_RULES_SOURCE_NAME,
+    {
+      refetchOnFocus: true,
+      refetchOnReconnect: true,
+      refetchOnMountOrArgChange: true,
+    }
+  );
+
+  const contactPoints = alertmanagerConfiguration.data?.alertmanager_config?.receivers ?? [];
+  return contactPoints;
+}
+
+function useGetDefaultContactPoint() {
+  const alertmanagerConfiguration = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useQuery(
+    GRAFANA_RULES_SOURCE_NAME,
+    {
+      refetchOnFocus: true,
+      refetchOnReconnect: true,
+      refetchOnMountOrArgChange: true,
+    }
+  );
+
+  return alertmanagerConfiguration.data?.alertmanager_config?.route?.receiver ?? '';
+}
+
+function useGetIncidentPluginConfig() {
+  const { installed: incidentPluginInstalled } = usePluginBridge(SupportedPlugin.Incident);
+  const [config, setConfig] = useState<IncidentsPluginConfig>({
+    isInstalled: false,
+    isChatOpsInstalled: false,
+    isIncidentCreated: false,
+  });
+
+  useEffect(() => {
+    if (!incidentPluginInstalled) {
+      setConfig({
+        isInstalled: false,
+        isChatOpsInstalled: false,
+        isIncidentCreated: false,
+      });
+      return;
+    }
+
+    getBackendSrv()
+      .post('/api/plugins/grafana-incident-app/resources/api/ConfigurationTrackerService.GetConfigurationTracker', {})
```

Comment:
The API expects a POST request even though we only fetch data?
Should we replace this with RTKQ?

Reply:
I also think it'd be good to have this in RTKQ, but equally it's a very specific part of a plugin, so I'm not sure how to best structure it. I'd say that it starts as something not generalised, and we improve the RTKQ API as we go. 

Perhaps it doesn't need to be done in this PR, could be done right afterwards (as there's already a lot of good changes in here :+1:)

Reply:
I thought the same 🤔

Reply:
I finally decided to use RTKQ since it's not a big task, and that way we maintain consistency in this PR code.

Reply:
I think the endpoint it's a POST because [of the Bearer TOKEN they use ](https://github.com/grafana/incident/blob/367a429c8afd774fb3842a31bbf968ce0674a663/api/oto/openapi.yaml)